index.js

```diff
@@ -4,15 +4,91 @@
 
   /*istanbul ignore next*/
   if(module && typeof module.exports !== 'undefined'){
-    module.exports = f();
+    module.exports = f(require('sanctuary-type-classes'), require('sanctuary-type-identifiers'));
   }else{
-    global.concurrify = f();
+    global.concurrify = f(global.sanctuaryTypeClasses, global.sanctuaryTypeIdentifiers);
   }
 
-}(/*istanbul ignore next*/(global || window || this), function(){
+}(/*istanbul ignore next*/(global || window || this), function(Z, type){
 
   'use strict';
 
-  return function(){};
+  var $alt = 'fantasy-land/alt';
+  var $ap = 'fantasy-land/ap';
+  var $map = 'fantasy-land/map';
+  var $of = 'fantasy-land/of';
+  var $zero = 'fantasy-land/zero';
+
+  var ordinal = ['first', 'second', 'third', 'fourth', 'fifth'];
+  var show = Z.toString;
+
+  function invalidArgument(it, at, expected, actual){
+    throw new TypeError(
+      it
+      + ' expects its '
+      + ordinal[at]
+      + ' ar gument to '
+      + expected
+      + '\n  Actual: '
+      + show(actual)
+    );
+  }
+
+  function invalidContext(it, actual){
+    throw new TypeError(
+      it + ' was invoked outside the context of a Concurrently. \n  Called on: ' + show(actual)
+    );
+  }
+
+  //concurrify :: Applicative m
+  //           => (TypeRep m, String, () -> m a, (m a, m a) -> m a, (m a, m (a -> b)) -> m b)
+  //           -> Concurrently m
+  return function concurrify(Repr, TYPEOF_CONCURRENTLY, zero, race, ap){
+
+    function Concurrently(sequential){
+      this.sequential = sequential;
+    }
+
+    function isConcurrently(x){
+      return x instanceof Concurrently || type(x) === TYPEOF_CONCURRENTLY;
+    }
+
+    function construct(x){
+      if(!Z.Functor.test(x)) invalidArgument('Concurrently', 0, 'to be a Functor', x);
+      if(!Z.Apply.test(x)) invalidArgument('Concurrently', 0, 'to be an Apply', x);
+      if(!Z.Applicative.test(x)) invalidArgument('Concurrently', 0, 'to be an Applicative', x);
```

Comment:
I'm not sure if I need to test explicitly for `Functor` and `Apply`. I think having Applicative there is enough since it depends on the others.